packages/rum/src/parentContexts.ts

```diff
@@ -58,6 +59,10 @@ export function startParentContexts(location: Location, lifeCycle: LifeCycle, se
     currentSessionId = session.getId()
   })
 
+  lifeCycle.subscribe(LifeCycleEventType.VIEW_UPDATED, (currentContext) => {
+    currentView = currentContext
+  })
```

Comment:
Instead of subscribing to view update, what about having a VIEW_COMPLETED event to update the location?

Reply:
The location changes during the view lifetime, not only when the view is completed.

Reply:
yes, we could keep to look at the actual location while the view is active.
Subscribe to view update coule would avoid to have two sources of truth though.

Reply:
I'd rather keep a single source of truth. If the VIEW_CREATED data was a View (containing the location), we could drop the 'location' usage altogether:
```diff
-url: (currentView!.location || location).href,
+url: currentView!.location.href,
```
WDYT?

Reply:
we could just add the location with id and start time

Reply:
That would introduce a new data structure instead of just re-using the View one. But alright.